src/compiler/checker.ts

```diff
@@ -11850,12 +11885,10 @@ namespace ts {
             // constructors of derived classes must contain at least one super call somewhere in their function body.
             const containingClassDecl = <ClassDeclaration>node.parent;
             if (getClassExtendsHeritageClauseElement(containingClassDecl)) {
-                const containingClassSymbol = getSymbolOfNode(containingClassDecl);
-                const containingClassInstanceType = <InterfaceType>getDeclaredTypeOfSymbol(containingClassSymbol);
-                const baseConstructorType = getBaseConstructorTypeOfClass(containingClassInstanceType);
+                const classExtendsNull = classDeclarationExtendsNull(containingClassDecl);
 
-                if (containsSuperCall(node.body)) {
-                    if (baseConstructorType === nullType) {
+                if (getSuperCallInConstructor(node)) {
+                    if (classExtendsNull) {
                         error(node, Diagnostics.A_constructor_cannot_contain_a_super_call_when_its_class_extends_null);

```

Comment:
Currently you're erroring on the constructor node itself, which gives a pretty unpleasant experience (it errors on the entire body as well).

You should report an error on the super call that you were able to find.